_posts/2018-05-08-pulp3-moving-to-rq.md

```diff
@@ -0,0 +1,145 @@
+archives for [March](https://www.redhat.com/archives/pulp-dev/2018-March/thread.html),
+[April](https://www.redhat.com/archives/pulp-dev/2018-April/thread.html), and
+[June](https://www.redhat.com/archives/pulp-dev/2018-June/thread.html).
+
+
+## User Changes
+
+### Redis, not RabbitMQ/Qpid
+
+RQ uses Redis as its backend, so with this change, Redis replaces RabbitMQ or Qpid in the Pulp
+architecture. Instead of running the workers with:
+
+### Run workers differently
+
+```commandline
+$ /path/to/python/bin/celery worker -A pulpcore.tasking.celery_app:celery -n resource_manager@%%h -Q resource_manager -c 1 --events --umask 18
+$ /path/to/python/bin/celery worker -A pulpcore.tasking.celery_app:celery -n reserved_resource_worker-1@%%h -Q reserved_resource_worker-1 -c  --events --umask 18
+```
+
+You run RQ workers with:
+
+```commandline
+rq worker -n 'resource_manager@%h' -w 'pulpcore.tasking.worker.PulpWorker'
+rq worker -n 'reserved_resource_worker_1@%h' -w 'pulpcore.tasking.worker.PulpWorker'
+```
+
+You'll want to update your process launcher scripts, e.g. systemd scripts to use these new worker
+commands instead.
+
+### Ensure DJANGO_SETTINGS is set
+
+Anywhere the worker is run, it explicitly needs the environment variable DJANGO_SETTINGS set. For
+example something like this in your environment or process launcher (e.g. systemd) definition:
+
+```commandline
+export DJANGO_SETTINGS_MODULE=pulpcore.app.settings
+``` 
+
+### Error handling differences
+
+All RQ workers exit when Redis is not available either on startup of a worker or if there is an
+existing connection. This is a behavioral difference from Celery. It is recommended that your
+process launcher, e.g. systemd be configured to wait-and-retry to restart the service if you want
+your RQ workers to automatically recover and reconnect when Redis has an availability issue.
+
+### TLS and Authentication
+
+If you are using TLS or authentication with RabbitMQ/Qpid, you'll need to port those to Redis. 
+
+
+## Plugin Writer Changes
+
+### Dispatching Tasks
+
+With Celery plugin writers would dispatch tasks using the `apply_async_with_reservation` method on
+the task object itself. It returns a `celery.result.AsyncResult`.
+
+```python
+import mytask
+result = mytask.apply_async_with_reservation([reservations], args=(...), kwargs={...})
+# result is a celery.result.AsyncResult instance tracking the task dispatched to the tasking system
+```
+
+With RQ, plugin writers dispatch tasks using the `enqueue_with_reservation` function provided by
+`pulpcore.tasking.tasks`. This takes any Python function as its first argument, with all subsequent
+arguments being similar to before. Here is an equivalent example:
+
+```python
+from pulpcore.tasking.tasks import enqueue_with_reservation
+import mytask
+result = enqueue_with_reservation(mytask, [reservations], args=(...), kwargs={...})
+# result is a rq.job.Job instance tracking the task dispatched to the tasking system
+```
+
+### Defining Tasks
+
```

Comment:
Did part of this section get left behind?  Also, hanging double backticks on line 121

Reply:
yup fixed in next push